### PR TITLE
fix user district

### PR DESCRIPTION
**Pull request by delcroip**

### CODE REVIEW

File: location/models.py [hunk 1]
```diff
@@ -271,11 +277,8 @@ def extend_allowed_locations(location_pks, strict=True, loc_types=None):
         logger.error(
             f"extend_allowed_locations is expecting a list but received {location_pks}"
         )
+    cache_location_graph_if_empty()
     graph = cache.get("location_graph")
```
Comment (dragos-dobre) [inline location/models.py:277]: I don't see this variable used. Do we need this?

File: location/models.py [hunk 2]
```diff
@@ -591,10 +594,8 @@ def get_user_districts(cls, user):
         cachedata = cache.get(f"user_districts_{user.id}")
         districts = []
         if cachedata is None:
+            cache_location_graph_if_empty()
             cache_district = cache.get("location_types")
```
Comment (dragos-dobre) [inline location/models.py]: Keep naming consistent with the content. Should be something like locations_by_type.

File: location/models.py [hunk 3]
```diff
@@ -591,10 +594,8 @@ def get_user_districts(cls, user):
...
-            if not cache_district:
-                cache_location_graph()
-                cache_district = cache.get("location_types")
             cachedata = []
             if user.is_superuser:
                 for loc in cache_district['D']:
```
Comment (dragos-dobre) [inline location/models.py]: Here will be locations_by_type['D'].

File: location/models.py [hunk 4]
```diff
@@ -617,7 +618,7 @@ def get_user_districts(cls, user):
                     .order_by("location__code")
                 )
             for d in districts:
```
Comment (dragos-dobre) [inline location/models.py:616]: Where do you populate districts?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: location/models.py (hunk 1 above)
Comment (delcroip) [reply location/models.py:277]: used line 290, you can rely on flake8 for this kind of things

File: location/models.py (hunk 2 above)
Comment (delcroip) [reply location/models.py]: No it does both type and graph

File: location/models.py (hunk 4 above)
Comment (delcroip) [reply location/models.py:616]: just above ..

**Pull request merged**